### DISCUSSION TITLE
Conditional Access Policy Blocks API use

### DISCUSSION THREAD

**Discussion by Rich-at-Chester-Business-Analytics** [migueesc123/PowerBIRESTAPI#175] [Q&A]
Good morning,

My company implemented a Conditional Access Policy as follows: 

Targets:  All Users
Target Resources (cloud apps):  Power BI, Power BI Service
Grant control:  Require device to be marked as compliant, Require Hybrid Azure AD joined device (Require one of these controls).

Once that was implemented I can no longer refresh my dataset using the API.  The error I get is Invalid Credentials.

Interestingly I can initiate a new connection, select a table, see a preview... but when I save the query I am asked to login and get the invalid credential error... so it appears that running from the query uses a different access method than creating/refreshing the query does (I'm guessing).

Any ideas on what I could modify in the policy to allow the API to continue to function?

Thank you!

RIch

**Comment by migueesc123** [comment]
I think this might go beyond what we can answer as the developers of the connector and this might fall more into what the Power BI REST API team would be able to answer. I'm converting this into a discussion hoping that other members of the community could potentially provide any insights into this.
